src/main/java/dev/blaauwendraad/masker/json/MaskingState.java

```diff
@@ -172,7 +172,7 @@ void backtrackCurrentJsonPath() {
     /**
      * Returns the TrieNode that references the end of the latest segment in the current jsonpath
      */
-    public KeyMatcher.TrieNode getCurrentJsonPathNode() {
+    public KeyMatcher.@Nullable TrieNode getCurrentJsonPathNode() {
```

Comment:
Woah, this looks so odd. Didn't even know this was possible in Java 😂

Reply:
Yeah, I didn't quite get why it needs to be done like that, I tried to do it on the whole field and it said something along the lines of "outer type can never be null for static inner class". While it kinda makes sense to apply it to inner type, I also don't get how that outer type can ever be null. I always thought that these outer types are kind of a namespace for static classes, you can import such class directly if you want, but can leave it be for more clarity (I.e. `Entry` vs `Map.Entry`), I always preferred the second, but it's so ugly with these annotations there 😅